sqliterepo/repository.c

```diff
@@ -691,7 +713,20 @@ __open_not_cached(struct open_args_s *args, struct sqlx_sqlite3_s **result)
 	sq3->admin = g_tree_new_full(metautils_strcmp3, NULL, g_free, g_free);
 
 	sqlx_exec(handle, "PRAGMA foreign_keys = OFF");
-	sqlx_exec(handle, "PRAGMA journal_mode = MEMORY");
+
+	/* We chose to check this call especially because it is able to detect
+	 * a wrong/corrupted database file. */
+	rc = sqlx_exec(handle, "PRAGMA journal_mode = MEMORY");
+	if (rc != SQLITE_OK) {
+		error = NEWERROR(CODE_INTERNAL_ERROR, "failed to setup base: (%d) %s",
```

Comment:
I would suggest returning a new specific `5XX` code when a corruption is explicitly detected.
Not that it is necessary immediately, but it might help to react on the error (from the client, the logs, etc)

Reply:
It is now `CODE_CORRUPT_DATABASE` (511).